front/src/modules/ui/table/editable-cell/type/components/GenericEditableNumberCellEditMode.tsx

```diff
@@ -31,12 +35,27 @@ export function GenericEditableNumberCellEditMode({ viewField }: OwnProps) {
     if (newText === fieldValue) return;
 
     try {
-      const numberValue = parseInt(newText);
+      let numberValue = parseInt(newText);
 
       if (isNaN(numberValue)) {
         throw new Error('Not a number');
       }
 
+      if (viewField.metadata.isPositive) {
+        if (!canBeCastAsPositiveIntegerOrNull(newText)) {
+          return;
+        }
+
+        const valueCastedAsPositiveNumberOrNull =
+          castAsPositiveIntegerOrNull(newText);
+
+        if (valueCastedAsPositiveNumberOrNull === null) {
+          throw Error('Not a number');
+        }
+
+        numberValue = valueCastedAsPositiveNumberOrNull;
+      }
+
       // TODO: find a way to store this better in DB
       if (numberValue > 2000000000) {
```

Comment:
We can move this check to canBeCastAsPositiveIntegerOrNull
Also let's use Number.MAX_SAFE_INTEGER instead of 2000000000

Reply:
Hey @Weiko, if we move this to canBeCastAsPositiveIntegerOrNull, we can't do this check for scenarios where isPositive flag is false. Please let me know your thoughts. Thank you

Reply:
Yes indeed, same as my other comment.

nit: You can still use Number.MAX_SAFE_INTEGER at least. We might want to store in DB (in viewFieldMetada) later